tools/build_defs/repo/cache.bzl

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# WARNING:
+# https://github.com/bazelbuild/bazel/issues/17713
+# .bzl files in this package (tools/build_defs/repo) are evaluated
+# in a Starlark environment without "@_builtins" injection, and must not refer
+# to symbols associated with build/workspace .bzl files
+
+visibility("private")
+
+NO_DEFAULT_CANONICAL_ID_ENV = "BAZEL_NO_DEFAULT_CANONICAL_ID"
```

Comment:
nit: slight prefer `BAZEL_NO_DEFAULT_CANONICAL_ID_FOR_DOWNLOAD` or something else more specific.

Reply:
I went with `BAZEL_HTTP_RULES_URLS_AS_DEFAULT_CANONICAL_ID`, do you think that's good?